nndownload/nndownload.py

```diff
@@ -2005,170 +2005,134 @@ def download_video_comments(
+        comments_limit = None
+    else:
+        output(f"Requesting up to {comments_limit} comments on each thread.\n", logging.INFO) # Defaults to 1000
+
+    with Progress(TextColumn("{task.description}"), BarColumn(), TaskProgressColumn(), TextColumn("{task.completed}/{task.total}"), transient=True) as progress:
+        tasks = []
         for thread in template_params["thread_params"]["targets"]:
-            thread_data = {
-                "commentCount": 0,
-                "comments": [],
-                "fork": thread["fork"],
-                "id": thread["id"]
-            }
-            COMMENTS_DATA_JSON["threads"].append(thread_data)
-
-            fetch_thread_comments(
-                session,
-                template_params["id"],
-                template_params["comment_server"],
-                template_params["thread_key"],
-                thread,
-                template_params["thread_params"]["language"],
-                last_time,
-                comments_limit,
-                thread_data
-            )
-        
-        output("Finished downloading comments for {0}.\n".format(template_params["id"]), logging.INFO)
+            comments_data = COMMENTS_DATA_JSON
+            thread = threading.Thread(target=fetch_comments_thread, args=(session, template_params["id"], template_params["comment_server"], template_params["thread_key"], thread, template_params["thread_params"]["language"], progress, comments_data, comments_from, comments_limit))
+            thread.start()
+            tasks.append({
+                "thread": thread,
+            })
+
+        for task in tasks:
+            task["thread"].join()
+
+        comments_data["globalComments"]["retrievedCount"] = sum(thread["retrievedCount"] for thread in comments_data["threads"])
+        comments_data["globalComments"]["commentCount"] = template_params["comment_count"]
 
-    except KeyboardInterrupt:
-        output("\nCTRL+C detected. Saving downloaded comments for {0}\n".format(template_params["id"]), logging.INFO)
-        raise
-    finally:
-        # Save in case of success and on interrupt)
         with open(filename, "w", encoding="utf-8") as file:
-            json.dump(COMMENTS_DATA_JSON, file, indent=None, ensure_ascii=False, sort_keys=True)
+            json.dump(comments_data, file, indent=None, ensure_ascii=False, sort_keys=True)
+
+        output("Saved comments for {0} to {1}.\n".format(template_params["id"], filename), logging.INFO)
+
 
-def fetch_thread_comments(
+def fetch_comments_thread(
     session: requests.Session,
     video_id: str,
     api_server: str,
     thread_key: str,
     thread: dict,
     language: str,
-    last_time: int,
-    comments_limit: int,
-    thread_data: dict
+    progress: Progress,
+    comments_data: dict,
+    comments_from: int,
+    comments_limit: int = None,
 ) -> None:
-    """Fetch comments for thread (requires login)."""
-    base_data = {
```

Comment:
This is really verbose. Collapsing the transient data like this into the request helps clarify what's happening